broadinstitute/single_cell_portal_core
# Upload sequence files in react upload wizard (SCP-3721)

SCP-3721.  Adding sequence file upload capability.  No validation yet -- that will come later.  I think we'll need a document summarizing all the validation rules we want to apply within and across forms

![image](https://user-images.githubusercontent.com/2800795/135372238-c19f9cd9-0e3d-49f1-af48-5f6005d30b7f.png)

 Also updates the way file ids are tracked after to address the bug Jon found where the bundle children would be reset after saving the parent.  The issue was the the id of the parent file changes when it's saved from `newFile-<x>` to a mongo GUID.  So I needed to make sure we were tracking the old id as well so the kids' bundle parent ids could be updated to the GUID after save

 Also addresses a comment from Eric in a previous PR about how `uploadUtils.js` was a bad filename in just about every way possible, so it's now split into two more sensibly organized files -- `upload-utils.js` for utility functions and `form-components.js` for small shared components.

TO TEST:
1 . Go to upload wizard for a study (make sure react upload wizard feature flag is on)
2. Go to sequence file step
3. add and save fastq file
4. add a file and set its type to BAM
5. set the species and assembly, and confirm the assembly menu updates with the species
6.  choose a BAI file
7.  confirm BAI file cannot be saved without first saving BAM file
8. save BAM file
9. observe BAI file and description are preserved
10. save BAI file
